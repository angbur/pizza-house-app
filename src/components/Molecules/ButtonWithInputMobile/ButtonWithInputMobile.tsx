import Button from 'components/Atoms/Button/Button';
import Icon from 'components/Atoms/Icon/Icon';
import { ChangeEventHandler, MouseEventHandler } from 'react';
import styled, { CSSProperties } from 'styled-components';

const StyledInput = styled.input`
  font-family: 'Source Sans Pro';
  font-size: 1rem;
  font-weight: 400;
  width: 40px;
  height: 43px;
  text-align: center;
  &:focus-visible {
    outline: none;
  }
  border-radius: 0;
  border: none;
  border-left: 1px solid black;
  border-right: 1px solid black;
`;

const StyledDiv = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  & > button {
    border-radius: 8px 0 0 8px;
  }
  & > input[type='number']::-webkit-inner-spin-button,
  & > input[type='number']::-webkit-outer-spin-button {
    -webkit-appearance: none;
    -moz-appearance: none;
    -os-appearance: none;
  }
  & > input[type='number'] {
    -moz-appearance: textfield;
  }
  @media (min-width: 545px) {
    display: none;
  }
`;

type ButtonWithInputMobileProps = {
  value: number;
  onIncrement?: MouseEventHandler<HTMLButtonElement>;
  onDecrement?: MouseEventHandler<HTMLButtonElement>;
  onClick?: MouseEventHandler<HTMLButtonElement>;
  onChange?: ChangeEventHandler<HTMLInputElement>;
  withRemove: boolean;
};

const ButtonWithInputMobile = ({
  value,
  onIncrement,
  onDecrement,
  onClick,
  onChange,
  withRemove,
}: ButtonWithInputMobileProps) => {
  const buttonStyles: CSSProperties = {
    border: 'none',
    maxWidth: '80px',
    maxHeight: '80px',
    borderRadius: '0',
  };
  return (
    <StyledDiv className={'quantity-button-mobile'}>
      <Button variant='primary-light' size={'md'} onClick={onIncrement} style={{ ...buttonStyles }}>
        {Icon({ name: 'plus', size: 20, isActive: false })}
      </Button>
      <StyledInput type='number' value={value} onChange={onChange} style={{ ...buttonStyles }} />
      <Button variant='primary-light' size={'md'} onClick={onDecrement} style={{ ...buttonStyles }}>
        {Icon({ name: 'minus', size: 20, isActive: false })}
      </Button>
      {withRemove && (
        <Button
          variant='primary-light'
          size={'md'}
          onClick={onClick}
          style={{ ...buttonStyles, borderLeft: '1px solid white' }}
        >
          {Icon({ name: 'trash', size: 20, isActive: false })}
        </Button>
      )}
    </StyledDiv>
  );
};

export default ButtonWithInputMobile;
