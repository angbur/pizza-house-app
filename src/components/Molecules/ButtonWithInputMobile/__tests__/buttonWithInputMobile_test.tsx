import renderer from 'react-test-renderer';
import ThemeProvider from 'components/Theme/ThemeContext';
import ButtonWithInputMobile from '../ButtonWithInputMobile';

const value = 1;

const Wrapper = () => {
  return (
    <ThemeProvider>
      <ButtonWithInputMobile value={value} withRemove={true} />
    </ThemeProvider>
  );
};

describe('Button with input for mobile', () => {
  test('should renders button with input correctly', () => {
    const tree = renderer.create(<Wrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
