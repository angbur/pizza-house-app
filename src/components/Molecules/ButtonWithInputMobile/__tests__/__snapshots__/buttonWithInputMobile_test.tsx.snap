// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button with input for mobile should renders button with input correctly 1`] = `
<div
  className="sc-kDvujY kpWPsJ quantity-button-mobile"
>
  <button
    className="sc-bcXHqe sc-gswNZR feNVH gHullA primary-light btn-md button"
    style={
      Object {
        "border": "none",
        "borderRadius": "0",
        "fontSize": "0.9500000000000001rem",
        "maxHeight": "80px",
        "maxWidth": "80px",
        "padding": "0.5850000000000001rem 1.125rem",
      }
    }
  >
    <img
      height={20}
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "plus.svg",
        }
      }
    />
  </button>
  <input
    className="sc-jrcTuL eyDxOv"
    style={
      Object {
        "border": "none",
        "borderRadius": "0",
        "maxHeight": "80px",
        "maxWidth": "80px",
      }
    }
    type="number"
    value={1}
  />
  <button
    className="sc-bcXHqe sc-gswNZR feNVH gHullA primary-light btn-md button"
    style={
      Object {
        "border": "none",
        "borderRadius": "0",
        "fontSize": "0.9500000000000001rem",
        "maxHeight": "80px",
        "maxWidth": "80px",
        "padding": "0.5850000000000001rem 1.125rem",
      }
    }
  >
    <img
      height={20}
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "minus.svg",
        }
      }
    />
  </button>
  <button
    className="sc-bcXHqe sc-gswNZR feNVH gHullA primary-light btn-md button"
    style={
      Object {
        "border": "none",
        "borderLeft": "1px solid white",
        "borderRadius": "0",
        "fontSize": "0.9500000000000001rem",
        "maxHeight": "80px",
        "maxWidth": "80px",
        "padding": "0.5850000000000001rem 1.125rem",
      }
    }
  >
    <img
      height={20}
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "trash.svg",
        }
      }
    />
  </button>
</div>
`;
